homeassistant/components/matter/entity.py

```diff
@@ -55,7 +57,129 @@ async def wrapper(self: MatterEntity, *args: P.args, **kwargs: P.kwargs) -> _R:
 
 
 @dataclass(frozen=True)
-class MatterEntityDescription(EntityDescription):
+class MatterEntityLabeling(EntityDescription):
+    """Data structure for Matter Tag and Label Information."""
+
+    # A label or tag can be used to modify an entity's name by appending the text
+    # or replacing the name. This class holds the information needed to do that.
+
+    # Where to place the Tag or Label text in the entity name.
+    # Can be "rename", "append", "ignore". Append by default.
+    class LabelPlacement(StrEnum):
+        """Enum for label placement options."""
+
+        APPEND = "append"
+        IGNORE = "ignore"
+        RENAME = "rename"
```

Comment:
It should be fine to instead place this outside of the `MatterEntityLabeling` class.

Reply:
No problem. Will take care of that this evening.